src/Stratis.Bitcoin.Features.Consensus/PowConsensusValidator.cs

```diff
@@ -313,16 +352,22 @@ public virtual void CheckBlockReward(ContextInformation context, Money nFees, Ch
             this.logger.LogTrace("(-)");
         }
 
-        protected virtual void CheckMaturity(UnspentOutputs coins, int nSpendHeight)
+        /// <summary>
+        /// Checks the maturity of UTXOs.
+        /// </summary>
+        /// <param name="coins">UTXOs.</param>
+        /// <param name="spendHeight">Height at which coins are considered spendable.</param>
```

Comment:
this seems incorrect, see the code CoinbaseMaturity is something like that you describe
this is something else, this seems to be the height at which the UTXOs are attempted to be spent